### DISCUSSION TITLE
Impossible to suppress 'Found encrypted data but no privateKey' errors in a specific situation

### DISCUSSION THREAD

**Discussion by ikesyo** [renovatebot/renovate#22181] [Suggest an Idea]
### Which Renovate are you using?

Both Renovate Open Source CLI (through [the GitHub Action](https://github.com/renovatebot/github-action)) and WhiteSource Renovate App

### Which platform are you using?

GitHub.com

### Have you checked the logs? Don't forget to include them if relevant

```
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "NoojEw2h1",
           "repository": "foo/bar",
           "msg": "Found encrypted data but no privateKey"
         },
         {
           "name": "renovate",
           "level": 50,
           "logContext": "NoojEw2h1",
           "repository": "foo/baz",
           "msg": "Found encrypted data but no privateKey"
         }
       ]
Error: The process '/usr/bin/docker' failed with exit code 1
```

### What would you like to do?

I'd like to suppress the errors in a situation:

- A repository is configured with both A) [the Renovate GitHub App](https://github.com/apps/renovate) and B) [the GitHub Action](https://github.com/renovatebot/github-action)
  - The latter (B) is for updating GitHub Actions versions (https://github.com/renovatebot/config-help/issues/570, https://github.com/renovatebot/renovate/issues/5733)
- The repository has `encrypted.npmToken` and `npmrc` options in its `renovate.json` config fifle
  - The `npmToken` is encrypted using https://app.renovatebot.com/encrypt so can be decrypted only in the Renovate GitHub App
  - That is used to access private npm modules (on A) but that is not needed when updating GitHub Actions (on B)
- Even though I have the following self-hosted config options when running CLI through the GitHub Action (on B), that doesn't take effect:
  - ```json
    "force": {
      "encrypted": null,
      "npmrc": null
    },
    ```

**Comment by rarkins** [comment]
We might need to add a special case, Eg if force.encrypted === null

**Comment by jarnove** [comment]
I'm having the same issue in the next scenario:

A repo with the renovate bot
Two other repositories that should be updated automatically, both configured with a fully encrypted npmrc file/value in renovate.json via **Encrypted entire .npmrc file into config**. (https://docs.renovatebot.com/private-modules/)

```json
"encrypted": {
  "npmrc": "..."
}
```

`Found encrypted data but no privateKey` on both repositories.

When I hardcode the npmrc file in the config file, it works, but that's of course not the best solution.

Any ideas?

**Comment by viceice** [comment]
@jahredhope You can't share encrypted renovate objects between github app and self-hosted bot.

If you like to use encryption with the saelf-hosted bot, you need to create a custom public and private key pair.

checkout https://docs.renovatebot.com/self-hosted-configuration/#privatekey

For more config questions please open a new discussion.
